File: frontend/src/types/analytics.ts
/** Analytics types. */

export interface AnalyticsSnapshot {
  id: string;
  community_id: string;
  snapshot_date: string;
  follower_count: number;
  engagement_rate: number;
  metric_metadata: Record<string, unknown> | null;
  created_at: string;
}

export interface AccountHealth {
  score: number;
  max_score: number;
  metrics: {
    total_reach?: number;
    new_subscribers?: number;
    engagement_rate?: number;
  };
}

export interface CommunityMetrics {
  id: string;
  name: string;
  platform: 'vk' | 'telegram';
  current_followers: number;
  follower_growth: number;
  engagement_rate: number;
  last_sync_at: string | null;
}

export interface SubscriberDynamics {
  period: string;
  data: Array<{
    date: string;
    vk: number;
    telegram: number;
  }>;
}

export interface DashboardData {
  account_health: AccountHealth;
  communities: CommunityMetrics[];
  subscriber_dynamics: SubscriberDynamics;
}

export interface Recommendation {
  id: string;
  type: string;
  priority: 'high' | 'medium' | 'low';
  title: string;
  description: string;
  action: {
    type: string;
    suggested_time?: string;
    platform?: 'vk' | 'telegram';
  };
}

File: backend/app/schemas/analytics.py
"""Analytics schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class AccountHealth(BaseModel):
    """Account health metrics."""

    score: int
    max_score: int
    metrics: dict[str, float]


class CommunityMetrics(BaseModel):
    """Community metrics."""

    id: UUID
    name: str
    platform: str
    current_followers: int
    follower_growth: int
    engagement_rate: float
    last_sync_at: datetime | None


class SubscriberDynamics(BaseModel):
    """Subscriber dynamics data."""

    period: str
    data: list[dict[str, int | str]]


class DashboardResponse(BaseModel):
    """Dashboard response schema."""

    account_health: AccountHealth
    communities: list[CommunityMetrics]
    subscriber_dynamics: SubscriberDynamics


class Recommendation(BaseModel):
    """Recommendation schema."""

    id: UUID
    type: str
    priority: str
    title: str
    description: str
    action: dict


class MetricValue(BaseModel):
    """Metric value with timestamp."""

    value: float
    recorded_at: datetime


class CommunityMetricDetail(BaseModel):
    """Detailed metric for a community."""

    metric_name: str
    values: list[MetricValue]
    trend: str  # 'up', 'down', 'stable'
    change_percent: float


class CommunityAnalyticsResponse(BaseModel):
    """Community analytics response schema."""

    community: dict[str, str | UUID]  # id, name, platform
    metrics: list[CommunityMetricDetail]
    period: dict[str, datetime]  # from, to


class RecommendationsResponse(BaseModel):
    """Recommendations response schema."""

    recommendations: list[Recommendation]
